### PR TITLE
[Improvement](planner) expand sql-block-rule to make it can be used  on all kinds of sql stmt 

### PR DESCRIPTION
**Pull request by Yulei-Yang**
Currently, sql-block-rule can only be used for query statements, while it's useful for other stmts like insert / delete / alter / drop etc. Now remove the limitation and expand its using scenario.

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
run feut

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateSqlBlockRuleStmt.java [hunk 1]
```diff
@@ -116,6 +118,11 @@ public void analyze(Analyzer analyzer) throws UserException {
         // check properties
         CreateSqlBlockRuleStmt.checkCommonProperties(properties);
         setProperties(properties);
+
+        // avoid a rule block any ddl for itself
+        if (StringUtils.isNotEmpty(sql) && Pattern.compile(sql).matcher(this.ruleName).find()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateSqlBlockRuleStmt.java:123]: why using `sql` to match the `ruleName`?

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 2]
```diff
@@ -468,6 +472,10 @@ private void executeByNereids(TUniqueId queryId) throws Exception {
         context.setStartTime();
         profile.getSummaryProfile().setQueryBeginTime();
         context.setStmtId(STMT_ID_GENERATOR.incrementAndGet());
+
+        if (checkBlockRules()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:476]: I think the `checkBlockRules()` no need to return a `bool` value?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateSqlBlockRuleStmt.java (hunk 1 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateSqlBlockRuleStmt.java:123]: if a sql_block_rule named block_rule_delete with sql "delete" is created, then any update to this block_rule will be blocked by itself, like "DROP SQL_BLOCK_RULE block_rule_delete", coz sql block rule name is a part of sql now.

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java (hunk 2 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:476]: fixed

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**